sdk/monitor/Azure.Monitor.Query/tests/LogsTestData.cs

```diff
@@ -111,26 +111,36 @@ private async Task InitializeData(string workspaceId, string workspaceKey)
                 return;
             }
 
-            while (count == 0)
+            while (!succeeded)
             {
                 await Task.Delay(TimeSpan.FromSeconds(30));
-                count = await QueryCount(workspaceId);
+                succeeded = await QueryCount(workspaceId);
             }
         }
 
-        private async Task<int> QueryCount(string workspaceId)
+        private async Task<bool> QueryCount(string workspaceId)
```

Comment:
Should we override IsEnvironmentReady for this logic?
https://grep.app/search?q=isenvironmentready